### PR TITLE
Adding new events and fixing some. 

### PR DESCRIPTION
**Pull request by weirded**
Improving behavior for messages with subtypes, as well as adding reactions.

### CODE REVIEW

File: src/main/scala/slack/models/package.scala [hunk 1]
```diff
@@ -155,6 +167,7 @@ package object models {
           case "channel_history_changed" => JsSuccess(jsValue.as[ChannelHistoryChanged])
           case "im_created" => JsSuccess(jsValue.as[ImCreated])
           case "im_opened" => JsSuccess(jsValue.as[ImOpened])
```
Comment (weirded) [inline src/main/scala/slack/models/package.scala]: I wasn't sure whether it was safe to remove this line, as the message comes in as `im_open`. Let me know and I'll clean it up.

Comment (gilbertw1) [inline src/main/scala/slack/models/package.scala]: The im_opened vs im_open was a typo, it can be safely removed

**Comment by gilbertw1** [comment]
Looks good to me! If you want to go ahead and remove that line, I'll merge it in

**Comment by weirded** [comment]
Ok, just removed the `im_opened` case. Should be ready to go.

**Comment by gilbertw1** [comment]
awesome....merged!

**Pull request merged**